### DISCUSSION TITLE
How  can I know a function is called by await?

### DISCUSSION THREAD

**Discussion by ahuangege** [nodejs/help#4873] [Q&A] [answered]
```
async function test2() {
    test();
    await test();
}

function test() {
    // I want to know if this is called by await ...
}
```

Can anybody help me? Thank You!

**Comment by targos** [comment, marked as answer]
You can't. `await` only has an effect on the value returned by the function. It's impossible to detect it from within the function.

**Reply by ahuangege** [reply]
Really bad news for me.

**Reply by aduh95** [reply]
You can mutate the return value itself though, if that's what you are looking for:
```js
function test(){
  return {
    get then(){this.wasAwaited = true;},
    wasAwaited: false
  };
}
console.log(test().wasAwaited); // false
console.log((await test()).wasAwaited) // true
```

**Reply by ahuangege** [reply]
No, it's not what i want. It's like this, server A sends a message to server B,  if it is await ,server B will response a message. If not await,  server B will not response.  So , I feel sad.

**Reply by satvik-1203** [reply]
It's impossible to tell if the API call is awaited or no, but I think you can do something like, inside ur body keep a key `isAwait` which is a boolean so depending on that server B will decide if it needs to send a res back or no. I hope that helps.

**Comment by ahuangege** [comment]
Thank you. I already did like what you say.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年8月14日(星期六) 晚上7:16
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [nodejs/node] How  can I know a function is called by await? (#39439)





 
I don't think there is a way to tell if the API call is awaited or no, but I think you can do something like, inside ur body keep a key isAwait which is a boolean so depending on that server B will decide if it needs to send a res back or no. I hope that helps.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by satvik-1203** [reply]
that's wonderful!

**Reply by satvik-1203** [reply]
also mark as answered so that other don't come and answer an answered question haha
